crates/uv/tests/tool_list.rs

```diff
@@ -85,3 +87,48 @@ fn tool_list_missing_receipt() {
     No tools installed
     "###);
 }
+
+#[test]
+fn tool_list_bad_environment() -> Result<()> {
+    let context = TestContext::new("3.12");
+    let tool_dir = context.temp_dir.child("tools");
+    let bin_dir = context.temp_dir.child("bin");
+
+    // Install `black`
+    context
+        .tool_install()
+        .arg("black==24.2.0")
+        .env("UV_TOOL_DIR", tool_dir.as_os_str())
+        .env("XDG_BIN_HOME", bin_dir.as_os_str())
+        .assert()
+        .success();
+
+    // Install `ruff`
+    context
+        .tool_install()
+        .arg("ruff==0.3.4")
+        .env("UV_TOOL_DIR", tool_dir.as_os_str())
+        .env("XDG_BIN_HOME", bin_dir.as_os_str())
+        .assert()
+        .success();
+
+    let venv_path = common::venv_bin_path(tool_dir.path().join("black"));
+    // Remove the python interpreter for black
+    fs::remove_dir_all(venv_path)?;
+
+    uv_snapshot!(context.filters(), context.tool_list()
+    .env("UV_TOOL_DIR", tool_dir.as_os_str())
+    .env("XDG_BIN_HOME", bin_dir.as_os_str()), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    ruff v0.3.4
+        ruff
+    
+    ----- stderr -----
+    warning: `uv tool list` is experimental and may change without warning.
+    Python interpreter not found at `[TEMP_DIR]/tools/black/bin/python3`
```

Comment:
Well now we fixed deletion but we're still going to snapshot the wrong path :) Is there a different way to break the environment?

Reply:
I'm actually trying to add a filter for the path but it's being kinda tricky

Reply:
```
    let venv_path = common::venv_bin_path(tool_dir.path().join("black"));
    // Remove the python interpreter for black
    fs::remove_dir_all(venv_path.clone())?;

    let venv_filter: Vec<(String, String)> = TestContext::path_patterns(&venv_path)
    .into_iter()
    .map(|pattern| (pattern, "[VENV_PATH]/".to_string()))
    .collect();

    let mut filters = context.filters().clone();
    filters.extend(
        venv_filter.iter().map(|(pattern, replacement)| (pattern.as_str(), replacement.as_str()))
    );
   uv_snapshot!(
          filters,
          ...
```

Reply:
shouldn't it work?

Reply:
it worked